standard/variables.md

```diff
@@ -987,3 +987,175 @@ variable_reference
+
+> *Example*: The following code shows examples of the different ref-safe-scopes. The declarations show the ref-safe-scope for a referent to be the initializing expression for a `ref` variable. The examples show the ref-safe-scope for a reference return:
+>
+> ```csharp
+> public class C
+> {
+>     // ref safe scope of arr is "caller-scope". 
+>     // ref safe scope of arr[i] is "caller-scope".
+>     private int[] arr = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 }; 
+> 
+>     public ref int M1(ref int r1) // ref safe scope is "caller-scope"
+>     {
+>         return ref r1; // r1 is safe to ref return
+>     }
+>
+>     public ref int M2(int v1) // ref safe scope is "function-member"
+>     {
+>         return ref v1; // error: v1 isn't safe to ref return
+>     }
+>
+>     public ref int M3()
+>     {
+>         int v2 = 5;
+> 
+>         return ref arr[v2]; // arr[v2] is safe to ref return
+>     }
+> 
+>     public void M4(int p) 
+>     {
+>         int v3 = 6;
+>         ref int r2 = ref p; // scope of r2 is block, ref safe scope of p is method
+>         ref int r3 = ref v3; // scope of r3 is block, ref safe scope of v3 is block
+>         ref int r4 = ref arr[v3]; // scope of r4 is block, ref safe scope of arr[v3] is caller method
+>     }
+> }
+> ```
+>
+> *end example.*
+
+#### §ref-span-safety-locals Local variable ref safe scope
+
+For a local variable `v`:
+
+- If `v` is a reference variable, its ref-safe-scope is the same as the ref-safe-scope of its initializing expression.
+- Otherwise its ref-safe-scope is the scope in which it was declared.
+
+#### §ref-span-safety-parameters Parameter ref safe scope
+
+For a formal parameter `p`:
+
+- If `p` is a `ref`, or `in` parameter, its ref-safe-scope is the caller-scope. It is safe-to-ref-return. If `p` is an `in` parameter, it can't be returned as a writable `ref` but can be returned as `ref readonly`.
+- If `p` is an `out` parameter, its ref-safe-scope is the function-member. It isn't safe-to-ref-return.
+- Otherwise, if `p` is the `this` parameter of a struct type, its ref-safe-scope is the function-member.
+- Otherwise, the parameter is a value parameter, and its ref-safe-scope is the function-member. It isn't safe-to-ref-return.
+
+#### §ref-span-safety-field-reference Field ref safe scope
+
+For a variable designating a reference to a field, `e.F`:
+
+- If `e` is of a reference type, its ref-safe-scope is the caller-scope.
+- Otherwise, if `e` is of a value type, its ref-safe-scope is the same as the ref-safe-scope of `e`.
+
+#### §ref-span-safety-operators Operators
+
+The conditional operator (§12.18), `c ? ref e1 : ref e2`, and reference assignment operator, `= ref e` (§12.21.1) have reference variables as operands and yield a reference variable. For those operators, the ref-safe-scope of the result is the narrowest scope among the ref-safe-scopes of all `ref` operands.
+
+#### §ref-span-safety-method-invocation Method invocation
```

Comment:
Does this apply to other function member invocations as well, such as indexer invocations?

Reply:
Resolved at the meeting: generalize to all function invocations (local functions, property and indexer accessors, and delegates).